runtime/queries/metricsview_aggregation.go

```diff
@@ -76,14 +90,159 @@ func (q *MetricsViewAggregation) Resolve(ctx context.Context, rt *runtime.Runtim
 		return fmt.Errorf("metrics view '%s' does not have a time dimension", q.MetricsView)
 	}
 
-	// Build query
-	sql, args, err := q.buildMetricsAggregationSQL(q.MetricsView, olap.Dialect(), q.ResolvedMVSecurity)
+	// execute druid query
+	sqlString, args, err := q.buildMetricsAggregationSQL(q.MetricsView, olap.Dialect(), q.ResolvedMVSecurity)
 	if err != nil {
 		return fmt.Errorf("error building query: %w", err)
 	}
 
-	// Execute
-	schema, data, err := olapQuery(ctx, olap, priority, sql, args)
+	if len(q.PivotOn) == 0 {
+		schema, data, err := olapQuery(ctx, olap, priority, sqlString, args)
+		if err != nil {
+			return err
+		}
+
+		q.Result = &runtimev1.MetricsViewAggregationResponse{
+			Schema: schema,
+			Data:   data,
+		}
+		return nil
+	}
+
+	if olap.Dialect() == drivers.DialectDuckDB {
+		return olap.WithConnection(ctx, priority, false, false, func(ctx context.Context, ensuredCtx context.Context, conn *databasesql.Conn) error {
+			temporaryTableName := tempName("_for_pivot_")
+
+			err = olap.Exec(ctx, &drivers.Statement{
+				Query:    fmt.Sprintf("CREATE TEMPORARY TABLE %s AS %s", temporaryTableName, sqlString),
+				Args:     args,
+				Priority: priority,
+			})
+			if err != nil {
+				return err
+			}
+			defer func() {
+				_ = olap.Exec(ensuredCtx, &drivers.Statement{
+					Query: `DROP TABLE "` + temporaryTableName + `"`,
+				})
+			}()
+
+			return q.pivotOn(ctx, olap, temporaryTableName)
+		})
+	}
+	dbOnce.Do(func() {
+		handle, errHandle = duckdbolap.Driver{}.Open(map[string]any{"pool_size": 10}, false, activity.NewNoopClient(), zap.NewNop())
+	})
+	if errHandle != nil {
+		return errHandle
+	}
+
+	schema, data, err := olapQuery(ctx, olap, priority, sqlString, args)
+	if err != nil {
+		return err
+	}
+
+	duckDBOLAP, _ := handle.AsOLAP("")
+	err = duckDBOLAP.WithConnection(ctx, priority, false, false, func(ctx context.Context, ensuredCtx context.Context, conn *databasesql.Conn) error {
+		temporaryTableName := tempName("_for_pivot_")
+		createTableSQL, err := duckdbolap.CreateTableQuery(schema, temporaryTableName)
```

Comment:
We should avoid using our DuckDB OLAP driver for this kind of ephemeral in-memory transformation. A few reasons:
- Our driver has a lot of extra overhead, like loading a bunch of extensions, migrations for catalog tables, emitting DB stats telemetry, etc.
- It leaks some pretty internal details of the driver into the `runtime/queries` package (like the `RawConn` func should not be public)
- Lifecycle management of the DuckDB handle becomes tricky with a global here, since DuckDB doesn't release memory it has taken. Since this is just for exports, it's probably better to open/close a handle on each call initially.

As discussed previously, I would suggest creating a generic `runtime/pkg/duckdbpivot` package with a `Pivot` function that takes a `sqlx.Rows` iterator as an argument, and returns a pivoted `sqlx.Rows`. Internally, it would use `database/sql` to directly open an in-memory DuckDB handle to just create the temporary table and do the pivot. 

That would also encapsulate the appender logic in a scoped package, and allow easier testing of the code.

===

An implication for the above is that here, the DuckDB and Druid cases become pretty different (since we want to avoid the overhead of converting from our DuckDB driver to `duckdbpivot`), but since the pivot SQL is pretty simple, I think some duplication is the best way to keep it simple.

Reply:
> Pivot function that takes a sqlx.Rows iterator as an argument, and returns a pivoted sqlx.Rows. Internally, it would use database/sql to directly open an in-memory DuckDB handle to just create the temporary table and do the pivot.

won't work. The function creates a handle but doesn't return it and it's not closed

Reply:
btw, I cannot pass `sqlx.Rows` because I have `drivers.Result` from the `OLAPStore`

Reply:
> Our driver has a lot of extra overhead, like loading a bunch of extensions, migrations for catalog tables, emitting DB stats telemetry, etc.
It leaks some pretty internal details of the driver into the runtime/queries package (like the RawConn func should not be public)
Lifecycle management of the DuckDB handle becomes tricky with a global here, since DuckDB doesn't release memory it has taken. Since this is just for exports, it's probably better to open/close a handle on each call initially.

Addressed those.